### PR TITLE
Support custom path for saving platform details

### PR DESCRIPTION
**Pull request by anandhu-eng**

### CODE REVIEW

File: script/detect-sudo/customize.py [hunk 1]
```diff
@@ -102,7 +104,7 @@ def prompt_sudo():
                 print(r.decode('utf-8'))  # Decode bytes to string
                 return 0
             except subprocess.TimeoutExpired:
-                reset_terminal()  # Reset terminal to sane state
+                reset_terminal()  # Reset terminal to same state
```
Comment (arjunsuresh) [inline script/detect-sudo/customize.py:107]: sane is correct :)

File: script/get-platform-details/_cm.json [hunk 2]
```diff
@@ -4,17 +4,44 @@
...
+  "input_mapping": {
+    "out_dir_path": "CM_PLATFORM_DETAILS_DIR_PATH",
+    "out_file_name": "CM_PLATFORM_DETAILS_FILE_NAME"
+  },
   "deps": [
     {
       "tags": "detect,os"
     },
     {
+      "skip_if_env": {
```
Comment (arjunsuresh) [inline script/get-platform-details/_cm.json:16]: We should not omit the dependency in this case. Instead, we can introduce run-macos and run-windows scripts within the get-platform-details module. Initially, these scripts can start as empty files. Over time, we can incrementally add commands that are specific to macOS and Windows to these scripts.

File: script/detect-sudo/customize.py (hunk 1 above)
Comment (anandhu-eng) [reply script/detect-sudo/customize.py]: corrected in [31723a8](https://github.com/mlcommons/cm4mlops/pull/418/commits/31723a83a049bee4bee1455e1fba02720c9276e6)

File: script/get-platform-details/customize.py [hunk 3]
```diff
@@ -13,17 +13,21 @@ def preprocess(i):
     os_info = i['os_info']
     env = i['env']
 
-    if os_info['platform'] == "windows":
-        return {'return':1, 'error':'get-platform-details script not yet supported in windows!'}
+    if os_info['platform'] == "darwin":
+        i['run_script_input']['script_name'] = "run-macos"
```
Comment (arjunsuresh) [inline script/get-platform-details/customize.py:17]: we don't need to do this - it happens automatically for different OS flavors

**Pull request merged**